### ISSUE TITLE
Potential Bugs: VectorDiffusionIntegrator::AssembleElementVector

### ISSUE THREAD

**Issue by hifabian** [mfem/mfem#4570]
I believe there are two bugs for `VectorDiffusionIntegrator` in the member function `AssembleElementVector`. Specifically one for `VectorCofficient` and one for `MatrixCoefficient` (though I only tested vector coefficients).

For vector coefficients the issue is the repeated multiplication of `w*vcoeff(k)` to `pelmat` in line 3093, see here:
https://github.com/mfem/mfem/blob/6105a4e1faa4a918a9cc5b7b6f0c86bc0f71177f/fem/bilininteg.cpp#L3088-L3098
A possible solution is to modify line 3096 and remove line 3093:
```c++
      if (VQ)
      {
         VQ->Eval(vcoeff, Tr, ip);
         for (int k = 0; k < vdim; ++k)
         {
            const Vector vec_in(mat_in.GetColumn(k), dof);
            Vector vec_out(mat_out.GetColumn(k), dof);
            pelmat.AddMult_a(w*vcoeff(k), vec_in, vec_out);
         }
      }
``` 

For matrix coefficients a similar error is made on line 3107 (and I believe line 3109 is also wrong), see here:
https://github.com/mfem/mfem/blob/6105a4e1faa4a918a9cc5b7b6f0c86bc0f71177f/fem/bilininteg.cpp#L3099-L3112
While I did not test this part, a solution should be:
```c++
      else if (MQ)
      {
         MQ->Eval(mcoeff, Tr, ip);
         for (int ii = 0; ii < vdim; ++ii)
         {
            Vector vec_out(mat_out.GetColumn(ii), dof);
            for (int jj = 0; jj < vdim; ++jj)
            {
               const Vector vec_in(mat_in.GetColumn(jj), dof);
               pelmat.AddMult_a(w*mcoeff(ii,jj), vec_in, vec_out);
            }
         }
      }
```

**Comment by vladotomov** [comment]
That's correct, thank you, please push your fix in #4572.

**Comment by hifabian** [comment]
I added the fixes and created a pull request (see #4573) to the relevant branch (bugfix-4750).

**Comment by vladotomov** [comment]
Continue in #4572.
